src/service-worker.js

```diff
@@ -0,0 +1,158 @@
+
+const mandatoryCacheList = [
+  '/',
+  '/styles.css',
+  '/offline.html',
+  '/custom-components/standard-page-wrapper.js'
+]
+
+const optionalCacheList = [
+  '/assets/images/alastair.260px.jpg',
+  '/assets/images/alastair.260px.webp',
+  '/assets/images/alastair.343px.jpg',
+  '/assets/images/alastair.343px.webp',
+  '/android-chrome-192x192.png',
+  '/android-chrome-512x512.png',
+  '/apple-touch-icon.png',
+  '/browserconfig.xml',
+  '/favicon-16x16.png',
+  '/favicon-32x32.png',
+  '/favicon.ico',
+  '/mstile-150x150.png',
+  '/safari-pinned-tab.svg',
+  '/site.webmanifest'
+]
+
+const addMandatoryItemsToCache = async () => {
+  const cache = await caches.open(LATEST_CACHE_NAME)
+  return cache.addAll(mandatoryCacheList)
+}
+
+const getFromLatestCache = async request => {
+  const cache = await caches.open(LATEST_CACHE_NAME)
+  return cache.match(request)
+}
+
+const getFromOldCache = async request => {
+  const cache = await caches.open(OLD_CONTENT_CACHE_NAME)
+  return cache.match(request)
+}
+
+const addToNewCache = async request => {
+  try {
+    const latestCache = await caches.open(LATEST_CACHE_NAME)
+    await latestCache.add(request)
+  } catch (error) {
+    console.log(`Failed to update new cache in background for ${request.url}`)
+  }
+}
+
+const updateCacheInBackground = getFromCache => async request => {
+  const response = await getFromCache(request)
+  if (response) addToNewCache(request)
+  return response
+}
+
+const fetchAndUpdateCache = async request => {
+  const cache = await caches.open(LATEST_CACHE_NAME)
+  const response = await fetch(request)
+  cache.put(request, response.clone())
+  return response
+}
+
+const isImage = request =>
+  request.url.endsWith('.png') ||
+  request.url.endsWith('.jpg') ||
+  request.url.endsWith('.webp')
+
+const getImageSize = key => Number(key.url.split('.')[-2].split('x')[0])
+
+const getBiggestImageFromCache = async (
+  cacheName,
+  requestedImageName,
+  requestedFileType
+) => {
+  const cache = await caches.open(cacheName)
+  const keys = await cache.keys()
+
+  const alternativeImages = keys.filter(key => {
+    const pathSections = (new URL(key.url)).pathname.split('.')
+    const imageName = pathSections.slice(0, -2).join('.')
+    const fileType = pathSections[-1]
+
+    return imageName === requestedImageName && fileType === requestedFileType
+  })
+  const biggestImage = alternativeImages
+    .sort((keyA, keyB) => getImageSize(keyB) - getImageSize(keyA))[0]
+  return biggestImage ? cache.match(biggestImage) : undefined
+}
+
+const getAlternativeImage = async request => {
+  const path = (new URL(request.url)).pathname
+  const pathSections = path.split('.')
+  const imageName = pathSections.slice(0, -2).join('.')
+  const fileType = pathSections[-1]
+
+  return getBiggestImageFromCache(LATEST_CACHE_NAME, imageName, fileType) ||
+    getBiggestImageFromCache(OLD_CONTENT_CACHE_NAME, imageName, fileType)
+}
+
+const getResource = async request => {
```

Comment:
Resources not in optionalCacheList or requiredCacheList should just be retrieved using `fetch`